site/src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,163 @@
+import {
+  ReactElement,
+  ReactNode,
+  cloneElement,
+  createContext,
+  useContext,
+  useRef,
+  useState,
+} from "react";
+// This is used as base for the main Popover component
+// eslint-disable-next-line no-restricted-imports -- Read above
+import MuiPopover, {
+  type PopoverProps as MuiPopoverProps,
+} from "@mui/material/Popover";
+
+type TriggerMode = "hover" | "click";
+
+type TriggerRef = React.RefObject<HTMLElement>;
+
+type TriggerElement = ReactElement<{
+  onClick?: () => void;
+  ref: TriggerRef;
+}>;
+
+type PopoverContextValue = {
+  open: boolean;
+  setOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  triggerRef: TriggerRef;
+  mode: TriggerMode;
+};
+
+const PopoverContext = createContext<PopoverContextValue | undefined>(
+  undefined,
+);
+
+export const Popover = (props: {
+  children: ReactNode | ((popover: PopoverContextValue) => ReactNode); // Allows inline usage
+  mode?: TriggerMode;
+  defaultOpen?: boolean;
+}) => {
+  const [open, setOpen] = useState(props.defaultOpen ?? false);
+  const triggerRef = useRef<HTMLElement>(null);
+  const value = { open, setOpen, triggerRef, mode: props.mode ?? "click" };
+
+  return (
+    <PopoverContext.Provider value={value}>
+      {typeof props.children === "function"
+        ? props.children(value)
+        : props.children}
+    </PopoverContext.Provider>
+  );
+};
+
+export const usePopover = () => {
+  const context = useContext(PopoverContext);
+  if (!context) {
+    throw new Error(
+      "Popover compound components cannot be rendered outside the Popover component",
+    );
+  }
+  return context;
+};
+
+export const PopoverTrigger = (props: {
+  children: TriggerElement;
+  hover?: boolean;
+}) => {
+  const popover = usePopover();
+
+  const clickProps = {
+    onClick: () => {
+      popover.setOpen((open) => !open);
+    },
+  };
+
+  const hoverProps = {
+    onMouseEnter: () => {
+      popover.setOpen(true);
+    },
+    onMouseLeave: () => {
+      popover.setOpen(false);
+    },
+  };
+
+  return cloneElement(props.children, {
+    ...(popover.mode === "click" ? clickProps : hoverProps),
+    ref: popover.triggerRef,
+  });
+};
+
+type Horizontal = "left" | "right";
+
+export const PopoverContent = (
+  props: Omit<MuiPopoverProps, "open" | "onClose" | "anchorEl"> & {
+    horizontal?: Horizontal;
+  },
+) => {
+  const popover = usePopover();
+  const horizontal = props.horizontal ?? "left";
+  const hoverMode = popover.mode === "hover";
+
+  return (
+    <MuiPopover
+      disablePortal
+      css={(theme) => ({
```

Comment:
I'm still not fully keyed into all of Emotion's nuances, so I'm not sure how hard this would be, but do you feel it makes sense to let components override the styles in bits and pieces?

If I'm reading this right, it looks like if a component decides to pass in a styling prop, even if it's just one property, that wipes all of the default styling

Reply:
Good question! It should be and it is done "automatically". The parent does not pass the `props.css` to its children but as `className`. I had this question as well on Slack when talking to Kayla.